assets/js/admin/orders.js

```diff
@@ -239,23 +250,128 @@ jQuery( document ).ready( ( $ ) => {
 		} );
 
 		$( document.body )
-			.off( 'wc_backbone_modal_response.facebook_for_commerce' )
-			.on( 'wc_backbone_modal_response.facebook_for_commerce', function() {
-				// copy the value of the modal select to the WC field
-				$( '#refund_reason' ).val( $( '#wc_facebook_refund_reason_modal' ).val() );
-				// submit the form
-				$form.data( 'allow-submit', true ).submit();
+		.off( 'wc_backbone_modal_response.facebook_for_commerce' )
+		.on( 'wc_backbone_modal_response.facebook_for_commerce', function () {
+			// copy the value of the modal select to the WC field
+			$( '#refund_reason' ).val( $( '#wc_facebook_refund_reason_modal' ).val() );
+			// submit the form
+			$form.data( 'allow-submit', true ).submit();
+		} );
+	}
+
+
+	/**
+	 * Displays the order complete modal on order form submit
+	 */
+	function displayCompleteModal() {
+
+		$( '#wc-backbone-modal-dialog .modal-close' ).trigger( 'click' );
+
+		if ( completeModalCarrierCode || completeModalTrackingNumber ) {
+			$( document.body )
+			.off( 'wc_backbone_modal_loaded' )
+			.on( 'wc_backbone_modal_loaded', function () {
+
+				if ( completeModalCarrierCode ) {
+					$( '#wc_facebook_carrier' ).val( completeModalCarrierCode );
+				}
+
+				if ( completeModalTrackingNumber ) {
+					$( '#wc_facebook_tracking_number' ).val( completeModalTrackingNumber );
+
+				}
 			} );
+		}
+
+		new $.WCBackboneModal.View( {
+			target: 'facebook-for-woocommerce-modal',
+			string: {
+				message: wc_facebook_commerce_orders.complete_modal_message,
+				buttons: wc_facebook_commerce_orders.complete_modal_buttons
+			}
+		} );
+
+		// handle confirm action
+		$( '.facebook-for-woocommerce-modal #btn-ok' )
+		.off( 'click.facebook_for_commerce' )
+		.on( 'click.facebook_for_commerce', ( event ) => {
+
+			event.preventDefault();
+			event.stopPropagation();
+
+			completeModalCarrierCode    = $( '#wc_facebook_carrier' ).val();
+			completeModalTrackingNumber = $( '#wc_facebook_tracking_number' ).val();
+
+			makeCompleteAjaxRequest( true, completeModalCarrierCode, completeModalTrackingNumber );
+		} );
 	}
 
 
-	$form.on( 'submit', function( event ) {
+	/**
+	 * Make complete order AJAX Request
+	 *
+	 * @param {Boolean} withModal
+	 * @param {String} trackingNumber
+	 * @param {String} carrierCode
+	 */
+	function makeCompleteAjaxRequest( withModal = false, trackingNumber = null, carrierCode = null ) {
```

Comment:
When the `trackingNumber` is empty, instead of a popup alert, there's an error being shown.

![XErDEnMWpi](https://user-images.githubusercontent.com/4955242/94311517-d3426880-ff51-11ea-891c-17b0f5b84ded.gif)

Reply:
Fixed in 868ff7b499d414fe2b646b108163bb2c6ac28a8f